custom-components/react-carousel/carousel.jsx

```diff
@@ -0,0 +1,110 @@
+const newId = (() => {
+  let uid = 0
+  return () => uid++
+})()
+
+function Product(props) {
+  const {Deeplink, ImageURL, Price, Title} = props.record
+  return (
+    <a href={Deeplink} className="product">
+      <img src={ImageURL} alt={Title} />
+      <div className="title">{Title}</div>
+      <div className="price">{Price}</div>
+    </a>
+  )
+}
+
+class Carousel extends React.Component {
```

Comment:
[Optional/open question]
Any particular reason to switch back to ES6 class instead of using functional components & hooks? Aren't hooks popular enough yet? In the previous version, there was neither a class nor `this` references, just regular functions. Kudos to React for making this possible.

Reply:
From what I read about it, as soon as you have state management it is preferrable to use class-based components. I can try to port it though, can be fun :-)

Reply:
No classes please 😄 
It was just this week when I was in  a discussion on class vs function in this context. It seemed the function approach also has quite a fanbase.

Reply:
Good! I'm on it